torch::optional

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23326 torch::Optional**

We have torch::List and torch::Dict and c10::optional to be used by custom operators.
This seems inconsistent. Let's expose c10::optional as torch::Optional to make naming more consistent.

Differential Revision: [D16464701](https://our.internmc.facebook.com/intern/diff/D16464701/)

smessmer committed:
torch::Optional

We have torch::List and torch::Dict and c10::optional to be used by custom operators.
This seems inconsistent. Let's expose c10::optional as torch::Optional to make naming more consistent.

Differential Revision: [D16464701](https://our.internmc.facebook.com/intern/diff/D16464701/)

colesbury reviewed (commented):
Putting optional in the torch namespace seems fine if it's part of the public API, but please don't change the capitalization

smessmer commented:
@colesbury Any specific reason why not? Seems more consistent with other types from the same API like `torch::List` and `torch::Dict`.

colesbury commented:
Matching the capitalization of List and Dict is less important than not introducing a new name for an existing widely used type.

smessmer committed:
Update on "torch::Optional"


We have torch::List and torch::Dict and c10::optional to be used by custom operators.
This seems inconsistent. Let's expose c10::optional as torch::Optional to make naming more consistent.

Differential Revision: [D16464701](https://our.internmc.facebook.com/intern/diff/D16464701/)

smessmer commented:
ok, changed it to lower case

smessmer commented:
nvm, torch::optional is already exported from torch/types.h
